### PR TITLE
Add rif loopback action test plan

### PR DESCRIPTION
**Pull request by nhe-NV**
Add test plan for the RIF interface loopback action feature.
The HLD for the RIF interface loopback action:  https://github.com/sonic-net/SONiC/blob/master/doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@prsunny could you please review or assign someone to review?

**Comment by liat-grozovik** [comment]
@liorghub please review 
@prsunny  could you please suggest someone to review and provide feedback?

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+### Supported topology
+The test will be supported on any topology
+
+
+## Test cases
+
+### Test cases #1 - Verify the loopback action can be configured and worked as expect,the interface type will cover all the 5 types of ip interface.
+1. Verify the default action with sending traffic: 
+   - Verify the traffic will be forwarded as expected.
+   - The RIF TX_ERR counter will not increase as expected
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:120]: Not all platforms support RIF counters, so can we check it generically. RIF or Interface TX_DRP

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+2. Configure the loopback action to drop on the interface with cli command
+3. Verify the the loopback action is configured correctly with show cli command.
+4. Send traffic to do the validation: 
+   - Verify the traffic will be dropped as expected.
+   - The RIF TX_ERR counter will increase as expected
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:125]: For some platforms it is part of Interface TX_DRP

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+1. Configure the loopback action on the interface, some to drop, some to forward.
+2. Disable the interfaces
+3. Enable the interfaces back
+4. Verify the loopback action is not changed with show cli command.
+5. Send traffic to do the validation: 
+   - Verify the traffic will be dropped/forwarded as expected.
+   - The RIF TX_ERR counter will increase on the interface that the loppback action is configured to drop
+   - The RIF TX_ERR counter will not increase on the interface that the loppback action is configured to forward
+
+### Test cases #3 - Verify loopback action is saved after reboot(config reload reboot, fast-reboot), the interface type will cover all the 5 types of ip interface.
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:142]: We would need one test case specifically to cover warmboot

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+1. Configure the loopback action on the interface, some to drop, some to forward.
+2. Config save -y
+3. Do reboot randomly(config reload/reboot/fast-reboot)
+4. Verify the loopback action is saved correctly with show cli command.
+5. Send traffic to do the validation: 
+   - Verify the traffic will be dropped as expected.
+   - The RIF TX_ERR counter will increase on the interface that the loppback action is configured to drop
```
Comment (prsunny) [inline doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:149]: loopback -> typo. please fix all places

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md (hunk 1 above)
Comment (nhe-NV) [reply doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:120]: Checked that the  TX_DRP in interface counter not increase as TX_ERR in the rif counter, do you have any suggestion for the platform not support RIF counters?
do we have any method to check if the platform support rif counter or not?

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md (hunk 3 above)
Comment (nhe-NV) [reply doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:142]: add warmboot type

File: doc/ip-interface/loopback-action/rif_loopback_action_testplan.md (hunk 4 above)
Comment (nhe-NV) [reply doc/ip-interface/loopback-action/rif_loopback_action_testplan.md:149]: done

**Comment by liat-grozovik** [comment]
after syncing with Prince, it was decided to keep it in this location. @prsunny  please approve following the comments fix.

**Comment by prsunny** [comment]
> after syncing with Prince, it was decided to keep it in this location. @prsunny please approve following the comments fix.

Hi Liat, after our discussion, I see @yxieca commented on this PR and requested to move to sonic-mgmt. @yxieca, if this is the plan, I suggest this also to be moved to be consistent and have a link in HLD to refer to the testplan 

https://github.com/sonic-net/SONiC/pull/1010#issuecomment-1178336594

**Comment by nhe-NV** [comment]
> > after syncing with Prince, it was decided to keep it in this location. @prsunny please approve following the comments fix.
> 
> Hi Liat, after our discussion, I see @yxieca commented on this PR and requested to move to sonic-mgmt. @yxieca, if this is the plan, I suggest this also to be moved to be consistent and have a link in HLD to refer to the testplan
> 
> [#1010 (comment)](https://github.com/sonic-net/SONiC/pull/1010#issuecomment-1178336594)

Moved it to the sonic-mgmt repo: https://github.com/Azure/sonic-mgmt/pull/5956
Close this one.

**Pull request closed without merging**